python/src/eval.cc

```diff
@@ -0,0 +1,89 @@
+#include "internal/eval.hh"
+#include "internal/errors.hh"
+#include "internal/nix-to-python.hh"
+#include "internal/python-to-nix.hh"
+#include <Python.h>
+
+#include <cxxabi.h>
+#include <store-api.hh>
+#include <finally.hh>
+
+namespace nix::python {
+
+const char * currentExceptionTypeName()
+{
+    int status;
+    auto res = abi::__cxa_demangle(abi::__cxa_current_exception_type()->name(), 0, 0, &status);
+    return res ? res : "(null)";
+}
+
+static PyObject * _eval(const std::string & expression, PyObject * argument)
+{
+    nix::Strings storePath;
+    nix::EvalState state(storePath, nix::openStore());
```

Comment:
Creating an `EvalState` for each call is wasteful and somewhat risky, because that's a less frequently used code path. While I would like to push the envelope on that, it's perhaps not the right to do that.
See https://github.com/tweag/nix/pull/17 for a global EvalState solution.